### DISCUSSION TITLE
Xbox button bind

### DISCUSSION THREAD

**Discussion by oleg-the-developer** [seerge/g-helper#1281] [Ideas]
Hey, would it be possible to add "xbox button" (the round one, with logo, number 4 on screenshot below) for a key bind ?
![image](https://github.com/seerge/g-helper/assets/68486078/8379efdd-b5a5-4f71-9a3c-ffe0fd01b9a4)
My understanding is that there is no keycode for this button specifically, so impossible to use 'custom' option. I think it would be useful for those who rely on steam big picture.

Thanks, let me know if i could help somehow.

**Comment by mozziemozz** [comment]
Hey, I just spent hours trying to figure this out lol. I originally wanted to map the guide button to the Command Center button (left below view button) but that doesn't work, like you said here.

However, I was able to select the Xbox Guide button as an action for any of the ROG buttons. I chose to map it to M1 since I don't really use the back paddles for anything else.

![image](https://github.com/user-attachments/assets/55707c79-de4d-42fb-8193-21d99c0ce991)

I hope it helps.

**Reply by seerge** [reply]
@mozziemozz hello,

You can't map Command Center or ROG / AC buttons to native controller actions, cause unfortunately (due to the way Asus created whole thing) they are not part of the XBox controller in a first place. They emit separate key-codes that G-Helper can capture, but then you can't force built in XBox controller to emit anything on software level.

But as you have correctly pointed out - all other buttons can be mapped. Cause this mapping happens on firmware level. G-Helper just sets what command should each button emit and then it will keep working even w/o G-Helper running.

I personally map `M + Menu` to emit Guide button, and `M+View` to emit `ALT+F4` (to close current app) :)

**Comment by mozziemozz** [comment]
@seerge thank you for the explanation. That makes sense. I'm very new to ROG Ally X. I only got it like 2 weeks ago and now clean installed W11 24H2 on it and discovered your awesome tool, so I don't need to use the Asus software.

I like M + Menu for Guide button, I think I'll do the same. Question about M1/M2: the UI shows primary and secondary actions but can they actually have two actions or is it just 1 + act as "shift" key? And is it always either of the M buttons or can it be specified which one triggers an action?

Also, I tried to use M + RB as Alt-Tab to cycle Windows but that doesn't work because the Alt Tab switcher disappears again if you let go and if you hold it, it cycles through apps way to fast. Would it be possible for you to add Win + Tab as action as well? That would more or less provide the same functionality, but you only have to press it once and then you can use arrow keys/DPAD/sticks to select an app and open it with A.

Note: I was able to achieve this by remapping Alt + Tab to Win + Tab by using PowerToys Shortcut Remapper but having it natively in G-Helper would be great.

**Reply by seerge** [reply]
All buttons except `M1` and `M2` can have both primary (when you press it directly) and secondary (when you press it with `M+` button together) actions. If you bind M to primary, it stops working as a trigger for secondary actions obviously. 

I'm not sure if `Win + Tab` is possible, but you can use `M + DPadDown` that shows desktop by default as [explained in readme](https://github.com/seerge/g-helper?tab=readme-ov-file#rog-ally-bindings)
